cpp/oneapi/dal/backend/interop/table_conversion.hpp

```diff
@@ -114,6 +115,105 @@ inline daal::data_management::NumericTablePtr convert_to_daal_table(const table&
     }
 }
 
+template <typename T>
+inline auto allocate_daal_csr_table(const T* data,
+                                    const std::size_t* column_indices,
+                                    const std::size_t* row_indices,
+                                    std::int64_t row_count,
+                                    std::int64_t column_count) {
+    return daal::data_management::CSRNumericTable::create(
+        data,
+        column_indices,
+        row_indices,
+        dal::detail::integral_cast<std::size_t>(column_count),
+        dal::detail::integral_cast<std::size_t>(row_count));
+}
+
+template <typename T>
+inline auto convert_to_daal_csr_table(array<T> data,
+                                      array<std::size_t> column_indices,
+                                      array<std::size_t> row_indices,
+                                      std::int64_t row_count,
+                                      std::int64_t column_count) {
+    if (!data.get_count() || !column_indices.get_count() || !row_indices.get_count())
+        return daal::services::SharedPtr<daal::data_management::CSRNumericTable>();
+    data.need_mutable_data();
+    column_indices.need_mutable_data();
+    row_indices.need_mutable_data();
+    const auto daal_data =
+        daal::services::SharedPtr<T>(data.get_mutable_data(), daal_object_owner{ data });
+    const auto daal_column_indices =
+        daal::services::SharedPtr<std::size_t>(column_indices.get_mutable_data(),
+                                               daal_object_owner{ column_indices });
+    const auto daal_row_indices =
+        daal::services::SharedPtr<std::size_t>(row_indices.get_mutable_data(),
+                                               daal_object_owner{ row_indices });
+
+    return daal::data_management::CSRNumericTable::create(
+        daal_data,
+        daal_column_indices,
+        daal_row_indices,
+        dal::detail::integral_cast<std::size_t>(column_count),
+        dal::detail::integral_cast<std::size_t>(row_count));
+}
+
+template <typename T>
+inline detail::csr_table convert_from_daal_csr_table(
+    const daal::data_management::CSRNumericTablePtr& nt) {
+    const std::int64_t row_count = nt->getNumberOfRows();
+    const std::int64_t column_count = nt->getNumberOfColumns();
+
+    T* daal_data{ nullptr };
+    std::size_t* daal_column_indices{ nullptr };
+    std::size_t* daal_row_indices{ nullptr };
+
+    nt->getArrays(&daal_data, &daal_column_indices, &daal_row_indices);
```

Comment:
In this case, when the daal table dies, the data will also die. Maybe return data through a shared pointer?